top level entity naming

superdiana opened the pull request.

slorello89 commented on opentok/opentok.py:
```diff
@@ -1127,3 +1130,26 @@ def _create_jwt_auth_header(self):
+        super(OpenTok, self).__init__(
+            api_key,
+            api_secret,
+            api_url="https://api.opentok.com",
```
named parameters here shouldn't be hardcoded in the superclass's constructor call. They are being passed into this constructor (and given a default value) so this would essentially be overriding any updated values the user passes in.

superdiana replied:
@slorello89  what do you mean? the `api_url` with values passed in here would overwrite what users write? a quick reference:

```
Python 3.7.6 (default, Dec 31 2019, 13:13:23) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def method(parameter='hola mundo'):
...     print(parameter)
... 
>>> method()
hola mundo
>>> method('adios mundo cruel')
adios mundo cruel
>>> 
```

slorello89 replied:
yeah, so in this case we're calling the parent class's constructor within our constructor - which is fine - however the api_url, which can be passed into our constructor - will be overridden by the hard coded http://api.opentok.com - because we're explicitly setting it as a named argument - my only suggestion here is to remove the hard coded string - which I think will override what the user passes in.

slorello89 reviewed (changes requested):
Looks good, except for the one comment that I left.

slorello89 reviewed (changes requested):
provided more feedback on the issue I pointed to before.

superdiana commented:
@slorello89 @dragonmantank In summary:

1. Renaming the OpenTok class to Client
2. An OpenTok subclass that inherits from the Client class to apply deprecation warnings. Fixed the parameter assignment on the super constructor
3. Modified the tests and sample to replace all the OpenTok instances for Client instances
4. Modified the Readme.rst to show how to init the OT client using the Client class

dragonmantank commented on README.rst:
```diff
@@ -39,13 +39,13 @@ Initializing
 
 Import the package at the top of any file where you will use it. At the very least you will need the
-``OpenTok`` class. Then initialize an OpenTok instance with your own API Key and API Secret.
+``Client`` class. Then initialize an Client instance with your own API Key and API Secret.
```
Grammatical error with the new text. Might be worthwhile to throw this through Grammarly just in case, I doubt any of this has every been properly copy-edited.

```suggestion
``Client`` class. Then initialize a Client instance with your own API Key and API Secret.
```

dragonmantank reviewed (changes requested):
Small grammatical error, but otherwise LGTM!

slorello89 reviewed (approved):
LGTM 👍
